src/Titon/Console/Command/AbstractCommand.hh

```diff
@@ -0,0 +1,344 @@
+        $this->input = $input;
+        $this->output = $output;
+        $this->arguments = new InputBag();
+        $this->flags = new InputBag();
+        $this->options = new InputBag();
+    }
+
+    /**
+     * Add a new `Argument` to be registered and parsed with the `Input`.
+     *
+     * @param \Titon\Console\InputDefinition\Argument $argument
+     *
+     * @return $this
+     */
+    public function addArgument(Argument $argument): this {
+        $this->arguments->set($argument->getName(), $argument);
+
+        return $this;
+    }
+
+    /**
+     * Add a new `Flag` to be registered and parsed with the `Input`.
+     *
+     * @param \Titon\Console\InputDefinition\Flag $flag
+     *
+     * @return $this
+     */
+    public function addFlag(Flag $flag): this {
+        $this->flags->set($flag->getName(), $flag);
+
+        return $this;
+    }
+
+    /**
+     * Add a new `Option` to be registered and parsed with the `Input`.
+     *
+     * @param \Titon\Console\InputDefinition\Option $option
+     *
+     * @return $this
+     */
+    public function addOption(Option $option): this {
+        $this->options->set($option->getName(), $option);
+
+        return $this;
+    }
+
+    /**
+     * Construct and return a new `Menu` object given the choices and display
+     * message.
+     *
+     * @param Map<mixed, mixed> $choices    Accepted values
+     * @param string $message               The message to display before the choices
+     *
+     * @return \Titon\Console\UserInput\Menu
+     */
+    protected function confirm(Map<mixed, mixed> $choices, string $message = ''): Menu {
+        $menu = new Menu($this->input, $this->output);
+        $menu->setAcceptedValues($choices)->setMessage($message);
+
+        return $menu;
+    }
+
+    /**
+     * Alias method for sending output through STDERROR.
+     *
+     * @param string $output    The message to send
+     */
+    protected function error(string $output): void {
+        $this->output->error($output);
+    }
+
+    /**
+     * Retrieve an `Argument` value by key.
+     *
+     * @param string $key
+     *
+     * @return mixed
+     */
+    protected function getArgument(string $key): ?string {
+        if ($argument = $this->input->getArgument($key)) {
+            return $argument->getValue();
+        }
+
+        return null;
+    }
+
+    /**
+     * Retrieve all `Argument` objects registered specifically to this command.
+     *
+     * @return \Titon\Console\InputBag<Argument>
+     */
+    public function getArguments(): InputBag<Argument> {
+        return $this->arguments;
+    }
+
+    /**
+     * Retrieve the command's description.
+     *
+     * @return string
+     */
+    public function getDescription(): string {
+        return $this->description;
+    }
+
+    /**
+     * Retrieve a `Flag` value by key.
+     *
+     * @param string $key
+     *
+     * @return mixed
+     */
+    protected function getFlag(string $key): ?int {
```

Comment:
Flags are integers since we can allow them to stack. i.e., -vvv (stackable set to true) will allow verbosity level to be 3.